ArchiSteamFarm/Bot.cs

```diff
@@ -2836,6 +2838,272 @@ public sealed class Bot : IAsyncDisposable {
 			}
 		}
 
+		private void OnInventoryChanged() {
+			Utilities.InBackground(CardsFarmer.OnNewItemsNotification);
+
+			if (BotConfig.BotBehaviour.HasFlag(BotConfig.EBotBehaviour.DismissInventoryNotifications)) {
+				Utilities.InBackground(ArchiWebHandler.MarkInventory);
+			}
+
+			if (BotConfig.SendSetsOnCompleted) {
+				Utilities.InBackground(SendCompletedSets);
+			}
+		}
+
+		private async Task<HashSet<uint>?> GetPossiblyCompletedBadgeAppIDs() {
+			using IDocument? badgePage = await ArchiWebHandler.GetBadgePage(1).ConfigureAwait(false);
+
+			if (badgePage == null) {
+				ArchiLogger.LogGenericWarning(Strings.WarningCouldNotCheckBadges);
+
+				return null;
+			}
+
+			byte maxPages = 1;
+			IElement? htmlNode = badgePage.SelectSingleNode("(//a[@class='pagelink'])[last()]");
+
+			if (htmlNode != null) {
+				string lastPage = htmlNode.TextContent;
+
+				if (string.IsNullOrEmpty(lastPage)) {
+					ArchiLogger.LogNullError(nameof(lastPage));
+
+					return null;
+				}
+
+				if (!byte.TryParse(lastPage, out maxPages) || (maxPages == 0)) {
+					ArchiLogger.LogNullError(nameof(maxPages));
+
+					return null;
+				}
+			}
+
+			HashSet<uint>? firstPageResult = GetPossiblyCompletedBadgeAppIDs(badgePage);
+
+			if ((firstPageResult == null) || (maxPages == 1)) {
+				return firstPageResult;
```

Comment:
This is logically correct, but hinders the purpose, if R# and other IDE checks do not yell about two ifs, one returning `null` and other returning `firstPageResult`, I'd prefer to split it that way. If they yell, it can stay like that.